File: Embedded_C_for_Spectrum_Digital_eZdsp_28335/generateSpectra.c
/*
 * generateSpectra.c
 *
 *  Created on: Apr 11, 2010
 *      Author: Jeff
 */
 #include "CableMonitoring.h"

// freqDomain = fft with linear ramp


void performFFT();

void generateSpectra() {
	int i;
	
    //Taking the Fourier Transform of input data
    performFFT(); // Perform FFT

    //Scale heights linearly with frequency
    for (i=0;i<SPECTRA_LENGTH;i++)
    {
        spectraTable[i]=i*(spectraTable[i]);
    }
}

File: Processing_M-Files/generateSpectra.m
% YfreqDomain = fft with linear ramp

%#eml

function YfreqDomain = generateSpectra(timeDomain, samplingHz) 
      
    %Taking the Fourier Transform of input data
    YfreqDomain=positiveFFT(timeDomain,samplingHz); % Perform FFT
    YfreqDomain = abs(YfreqDomain); % Convert from complex to real
    
    %Scale heights linearly with frequency
    for i=1:length(YfreqDomain),
        YfreqDomain(i)=(i*YfreqDomain(i));
    end
end